pkg/exporter/grpc_proto.go

```diff
@@ -21,19 +24,23 @@ type GRPCProto struct {
 	// maxFlowsPerMessage limits the maximum number of flows per GRPC message.
 	// If a message contains more flows than this number, the GRPC message will be split into
 	// multiple messages.
-	maxFlowsPerMessage int
+	maxFlowsPerMessage            int
+	numberOfRecordsReceivedByGRPC prometheus.Counter
+	errExportByGRPC               prometheus.Counter
 }
 
-func StartGRPCProto(hostIP string, hostPort int, maxFlowsPerMessage int) (*GRPCProto, error) {
+func StartGRPCProto(hostIP string, hostPort int, maxFlowsPerMessage int, m *metrics.Metrics) (*GRPCProto, error) {
 	clientConn, err := grpc.ConnectClient(hostIP, hostPort)
 	if err != nil {
 		return nil, err
 	}
 	return &GRPCProto{
-		hostIP:             hostIP,
-		hostPort:           hostPort,
-		clientConn:         clientConn,
-		maxFlowsPerMessage: maxFlowsPerMessage,
+		hostIP:                        hostIP,
+		hostPort:                      hostPort,
+		clientConn:                    clientConn,
+		maxFlowsPerMessage:            maxFlowsPerMessage,
+		numberOfRecordsReceivedByGRPC: m.CreateNumberOfRecordsReceivedByGRPC("number_of_records_received_by_grpc"),
```

Comment:
What about a counter named `records_written_total` with a label such as `transport` or `exporter` that could be either `grpc` or `kafka` or anything else?
Also it may be useful to count records but also the number of batches, so there could be a second metric `batches_written_total` with again an `exporter` or `transport` label.

On naming metrics and labels, it's recommend to read this: https://prometheus.io/docs/practices/naming/

Reply:
line 52 just after `for inputRecords := range input {` if you increment a batch counter by 1, wouldn't it give the batch count?